src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkMigrationService.kt

```diff
@@ -0,0 +1,103 @@
+package uk.gov.justice.digital.hmpps.whereabouts.services.court
+
+import jakarta.persistence.EntityNotFoundException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.whereabouts.dto.AppointmentLocationTimeSlot
+import uk.gov.justice.digital.hmpps.whereabouts.dto.VideoBookingMigrateEvent
+import uk.gov.justice.digital.hmpps.whereabouts.dto.VideoBookingMigrateResponse
+import uk.gov.justice.digital.hmpps.whereabouts.model.HearingType
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkAppointment
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBookingEvent
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkAppointmentRepository
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingEventRepository
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingRepository
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Service
+class VideoLinkMigrationService(
+  private val videoLinkBookingRepository: VideoLinkBookingRepository,
+  private val videoLinkBookingEventRepository: VideoLinkBookingEventRepository,
+  private val videoLinkAppointmentRepository: VideoLinkAppointmentRepository,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getVideoLinkBookingToMigrate(videoBookingId: Long): VideoBookingMigrateResponse {
+    val videoLinkBooking = videoLinkBookingRepository.findById(videoBookingId)
+      .orElseThrow { EntityNotFoundException("Video link booking ID {videoBookingId} was not found") }
+
+    log.info("Migrating video link booking ID $videoBookingId")
+
+    val appointments = videoLinkAppointmentRepository.findAllByVideoLinkBooking(videoLinkBooking)
+    require(appointments.isNotEmpty()) { "Video link booking ID $videoBookingId has no appointments" }
```

Comment:
I would be inclined to verify the length of the `appointments` as well i.e. <= 3.  Should never be more than 3.